tests/unittests/test_gcp_authorization.py

```diff
@@ -1,125 +1,110 @@
-from typing import Optional
+from unittest.mock import Mock, patch
 
+import google.auth.exceptions
 import pytest
-from unittest.mock import Mock, patch
-from google.auth.credentials import Credentials
-from google.auth.exceptions import RefreshError
+from google.auth.transport.requests import Request
 
 from ai21.clients.vertex.gcp_authorization import GCPAuthorization
-from google.auth.transport.requests import Request
+from ai21.errors import CredentialsError
 
 _TEST_PROJECT_ID = "test-project"
 
 
-@pytest.mark.parametrize(
-    ids=[
-        "when_project_id_not_found__should_raise_error",
-        "when_project_id_found__should_set_auth",
-    ],
-    argvalues=[(None,), (_TEST_PROJECT_ID,)],
-    argnames=["project_id"],
-)
 @patch("google.auth.default")
 @patch.object(GCPAuthorization, GCPAuthorization._get_gcp_request.__name__)
-def test__gcp_authorization_set_auth(
+def test__get_gcp_credentials_and_project_id__when_project_id_not_found__should_raise_error(
     mock_gcp_request: Mock,
     mock_default: Mock,
     mock_gcp_credentials: Mock,
-    project_id: Optional[str],
 ):
-    mock_default.return_value = (mock_gcp_credentials, project_id)
+    mock_default.return_value = (mock_gcp_credentials, None)
     mock_gcp_request.return_value = Mock(spec=Request)
 
-    auth = GCPAuthorization(project_id=project_id)
+    auth = GCPAuthorization()
+
+    with pytest.raises(ValueError):
+        _, _ = auth.get_gcp_credentials_and_project_id(project_id=None)
```

Comment:
No need for `_, _ =`
A call to the function without loading it into variables, is fine as well since you're expecting an error